File: daily/14/solution.js
//题目来源： https://leetcode-cn.com/problems/longest-common-prefix/

var longestCommonPrefix = function(strs) {
    var n = strs.length;
    if(n === 0) return "";
    if(n === 1) return strs[0];


    var ans = strs[0];
    for(var i = 1; i < n; i++) {
        var str = strs[i];
        var j = 0;
        var index = 0;
        while(j < ans.length && j < str.length) {
            if(ans[j] !== str[j++]) break;
            else index++;

        }
        ans = ans.substring(0, index);
    }

    return ans;
};

File: daily/14/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(!strs.size()) return "";

        string prefix = strs[0];
        int count = strs.size();
        for(int i = 0; i < count; ++i){
            prefix = func(prefix, strs[i]);
            if(!prefix.size()) break;
        }
        return prefix;
        
    }
    string func(const string& str1, const string& str2){
        int length = min(str1.size(), str2.size());
        int index = 0;
        while(index < length && str1[index] == str2[index]) ++index;
        return str1.substr(0, index);
    }
};